Documentation for Pan and Zoom events (#2065)
telerik/blazor-docs: components/map/events.md

* [OnClick](#onclick)
* [OnMarkerClick](#onmarkerclick)
* [OnShapeClick](#onshapeclick)
* [OnZoomEnd](#onzoomend)
* [OnPanEnd](#onpanend)

## OnClick

> The `OnClick` event fires when the user clicks or taps on the Map. The `OnClick` event handler receives an argument of type `MapClickEventArgs`, which exposes the following properties:

Reviewer: ```suggestion
The `OnClick` event fires when the user clicks or taps on the Map. The `OnClick` event handler argument is of type `MapClickEventArgs`, which exposes the following properties:
```

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Property | Type | Description |
| ---------| ---- | ----------- |

> | `EventArgs` | `EventArgs` |  The native DOM event (browser event). |

Reviewer: If this is the `MouseEventArgs`, then you can clarify it:

```suggestion
| `EventArgs` | `EventArgs` |  The properties of the native browser event. Cast it to [`MouseEventArgs`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.web.mouseeventargs). |
```

> | `Location` | `MapLocation` | The location of the click on the Map (`MapLocation` has `Latitude` and `Longitude` props). |

Reviewer: ```suggestion
| `Location` | `MapLocation` | The location of the click on the Map (`MapLocation` has `Latitude` and `Longitude` properties). |
```

## OnMarkerClick

> The `OnMarkerClick` event fires when the user clicks on a marker. The `OnMarkerClick` event handler receives an argument of type `MapMarkerClickEventArgs`, which exposes the following properties:

Reviewer 2: Usually, it's "click something" rather than "click on something". The only exception throughout this article would be the `OnClick` section, where "click on the map" implies that the user clicks anywhere on/within the map. And if you decide to go with `Map` as the name of the component, you can also use `clicks the Map` (you're using Map below). 

In all other instances, (i.e., "clicks a marker", "clicks a shape") would be the better choice. Refer to the [Microsoft Manual of Style](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/term-collections/mouse-mouse-interaction-terms), since it seems we haven't included this as an explicit point in our style guide (although, it's the prevalent form in our examples).

I would also suggest (as per the MMoS recs) to user more specific verbs. For example, if the same event fires when the user taps something on a touch-enabled device, you might want to use `clicks or taps X`.

## OnShapeClick

The `OnShapeClick` event fires when the user clicks or taps a shape. The `OnShapeClick` event handler receives an argument of type `MapShapeClickEventArgs`, which exposes the following properties:

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Property | Type | Description |
| ---------| ---- | ----------- |
| `EventArgs` | `EventArgs` | The native DOM event (browser event). |

> | `DataItem` | `object` | The data item when the shape is coming from a Bubble layer (`null` for Shape layer). |

Reviewer: ```suggestion
| `DataItem` | `object` | The data item when the shape is from a Bubble layer, or `null` when the shape is from a Shape layer. |
```

            $"clientX = {eventArgs.ClientX}, clientY = {eventArgs.ClientY}");
    }

    private void LogToConsole(string text)
    {
        EventResult = text;
    }

    public class MarkerModel
    {
        public double[] LatLng { get; set; }
        public string Title { get; set; }
    }

    public class BubbleModel
    {
        public double[] LatLng { get; set; }
        public int Revenue { get; set; }
    }
}
````

## OnZoomEnd 

> The `OnZoomEnd` event fires when the user has finished zooming the Map. The `OnZoomEnd` event handler receives an argument of type `MapZoomEndEventArgs`, which exposes the following properties:

Reviewer 2: Double-check your use of map vs Map here, in onPanEnd and in the OnClick section. Apply consistency as needed.

            LatLng = new double[] { 41.8781, -87.6298 },
            Revenue = 200
        }
    };

    private void OnZoomEnd(MapZoomEndEventArgs args)
    {
        var zoom = args.Zoom;
        var center = args.Center;
        var extent = args.Extent;

        LogToConsole(
            $"zoom end: zoom level = {zoom}, center coordinates = [{center[0]},{center[1]}]," +
            $"extent = NW - [{extent[0]}, {extent[1]}]; SE - [{extent[2]}, {extent[3]}]");
    }

    private void LogToConsole(string text)
    {
        EventResult = text;
    }

    public class MarkerModel
    {
        public double[] LatLng { get; set; }
        public string Title { get; set; }
    }

    public class BubbleModel
    {
        public double[] LatLng { get; set; }
        public int Revenue { get; set; }
    }
}
````

## OnPanEnd

The `OnPanEnd` event fires when the user has finished moving (panning) the Map. The `OnPanEnd` event handler receives an argument of type `MapPanEndEventArgs` argument, which exposes the following properties:

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Property | Type | Description |
| ---------| ---- | ----------- |

> | `Center` | `double[]` | The lat / lang of the Map's center. |

Reviewer 2: Expand the contractions to their complete forms (i.e., latitude and longitude) in both table rows.

| `Extent` | `double[]` | The NW and SE latitude and longitude of the Map. |

> >caption Handle OnPanEnd.

Reviewer: No dot and more keywords

```suggestion
>caption Handle the Map OnPanEnd event
```
